src/zero_copy/container.rs

```diff
@@ -0,0 +1,456 @@
+        T: PartialEq,
+    {
+        <[T]>::contains(self, val)
+    }
+}
+
+impl<'b, T> Seq<T> for LinkedList<T> {
+    type Item<'a> = &'a T
+    where
+        Self: 'a;
+
+    type Iter<'a> = alloc::collections::linked_list::Iter<'a, T>
+    where
+        Self: 'a;
+
+    fn seq_iter(&self) -> Self::Iter<'_> {
+        self.iter()
+    }
+
+    fn contains(&self, val: &T) -> bool
+    where
+        T: PartialEq,
+    {
+        LinkedList::contains(self, val)
+    }
+}
+
+impl<T: Eq + Hash> Seq<T> for HashSet<T> {
+    type Item<'a> = &'a T
+    where
+        Self: 'a;
+
+    type Iter<'a> = hashbrown::hash_set::Iter<'a, T>
+    where
+        Self: 'a;
+
+    fn seq_iter(&self) -> Self::Iter<'_> {
+        self.iter()
+    }
+
+    fn contains(&self, val: &T) -> bool
+    where
+        T: PartialEq,
+    {
+        HashSet::contains(self, val)
+    }
+}
+
+#[cfg(feature = "std")]
+impl<T: Eq + Hash> Seq<T> for std::collections::HashSet<T> {
+    type Item<'a> = &'a T
+    where
+        Self: 'a;
+
+    type Iter<'a> = std::collections::hash_set::Iter<'a, T>
+    where
+        Self: 'a;
+
+    fn seq_iter(&self) -> Self::Iter<'_> {
+        self.iter()
+    }
+
+    fn contains(&self, val: &T) -> bool
+    where
+        T: PartialEq,
+    {
+        self.contains(val)
+    }
+}
+
+impl<T: Ord> Seq<T> for alloc::collections::BTreeSet<T> {
+    type Item<'a> = &'a T
+    where
+        Self: 'a;
+
+    type Iter<'a> = alloc::collections::btree_set::Iter<'a, T>
+    where
+        Self: 'a;
+
+    fn seq_iter(&self) -> Self::Iter<'_> {
+        self.iter()
+    }
+
+    fn contains(&self, val: &T) -> bool
+    where
+        T: PartialEq,
+    {
+        self.contains(val)
+    }
+}
+
+macro_rules! impl_for_range {
+    ($($ty:ty),* $(,)?) => {
+        $(
+        impl Seq<$ty> for Range<$ty> {
+            type Item<'a> = $ty
+            where
+                Self: 'a;
+
+            type Iter<'a> = Range<$ty>
+            where
+                Self: 'a;
+
+            fn seq_iter(&self) -> Self::Iter<'_> {
+                self.clone()
+            }
+
+            fn contains(&self, val: &$ty) -> bool {
+                Range::contains(self, val)
+            }
+        }
+
+        impl Seq<$ty> for core::ops::RangeInclusive<$ty> {
+            type Item<'a> = $ty
+            where
+                Self: 'a;
+
+            type Iter<'a> = core::ops::RangeInclusive<$ty>
+            where
+                Self: 'a;
+
+            fn seq_iter(&self) -> Self::Iter<'_> {
+                self.clone()
+            }
+
+            fn contains(&self, val: &$ty) -> bool {
+                core::ops::RangeInclusive::contains(self, val)
+            }
+        }
+
+        // TODO: Other range types
+
+        impl OrderedSeq<$ty> for Range<$ty> {}
+        impl OrderedSeq<$ty> for core::ops::RangeInclusive<$ty> {}
+        )*
+    }
+}
+
+impl_for_range!(
+    u8, u16, u32, u64, usize,
+    i8, i16, i32, i64, isize,
+    char,
+);
```

Comment:
Can we not just generalise the impl to the bounds already on [`contains`](https://doc.rust-lang.org/std/ops/struct.Range.html#method.contains) ?

Reply:
This is for the iterable part of ranges - they require `Step`, which is unstable. I could make `iter` part of `OrderedSeq`, so 'unordered' just lets you check bounds, then 'ordered' requires iteration, then it would just be a macro for the ordered parts. (I could also make them not ordered. That seems a bit disingenous though, as while they're not necessarily a common use case for `just`, they do have a strong order)

Reply:
That's unfortunate. Perhaps add a `// TODO: Generalise this with 'Step' later` to the code?

Reply:
The step issue provided me with a workaround I feel silly for not considering: `where Self: Iterator`